sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -1186,6 +1241,16 @@ static const struct platform_device_id board_ids[] = {
 					SOF_RT5682_SSP_AMP(0) |
 					SOF_RT5682_NUM_HDMIDEV(3)),
 	},
+	{
+		.name = "adl_rt5682_c1_h02",
+		.driver_data = (kernel_ulong_t)(SOF_RT5682_MCLK_EN |
+					SOF_RT5682_SSP_CODEC(1) |
+					SOF_RT5682_NUM_HDMIDEV(3) |
+					SOF_NO_OF_HDMI_CAPTURE_SSP(2) |
+					SOF_HDMI_CAPTURE_1_SSP(0) |
+					SOF_HDMI_CAPTURE_2_SSP(2) |
+					SOF_SSP_HDMI_CAPTURE_PRESENT),
+	},
```

Comment:
why would you put this last instead of grouping it with the rest of ADL stuff? This is just too messy.

Reply:
Agree. this messed up  happen due to changes brought from 5.15 development kernel into mainline. Actually there is no RPL, MTL support on 5.15 kernel and no entry for other boards and its last entry. 
We will correct it and update.